disco/preprocess/select_timepoints2.py

```diff
@@ -5,154 +5,310 @@
         if name in used_profiles:
-            if 'active' in profile_types:
+            if "active" in profile_types:
                 original_p_mult = dss.LoadShape.PMult()
-                
-            if 'reactive' in profile_types:
+
+            if "reactive" in profile_types:
                 original_q_mult = dss.LoadShape.QMult()
-                
+
             dss.LoadShape.Npts(number_of_timepoints)
-            
-            if len(original_p_mult)>1:
-                if len(original_p_mult)>max(critical_time_indices):
+
+            if original_p_mult is not None and len(original_p_mult) > 1:
+                if len(original_p_mult) > max(critical_time_indices):
                     dss.LoadShape.PMult(list(np.array(original_p_mult)[critical_time_indices]))
                 else:
-                    raise Exception("IndexError: Index out of range") 
-            if len(original_q_mult)>1:
-                
-                if len(original_q_mult)>max(critical_time_indices):
+                    raise Exception("IndexError: Index out of range")
+            if original_q_mult is not None and len(original_q_mult) > 1:
+                if len(original_q_mult) > max(critical_time_indices):
                     dss.LoadShape.QMult(list(np.array(original_q_mult)[critical_time_indices]))
                 else:
-                    raise Exception("IndexError: Index out of range") 
+                    raise Exception("IndexError: Index out of range")
         flag = dss.LoadShape.Next()
 
-def save_circuit(output_folder):
-    dss.Text.Command(f'Save Circuit Dir={output_folder}')
-    
-            
+
+def save_circuit(output_folder: Path):
+    """Run the OpenDSS command to save a compiled circuit into a directory."""
+    dss.Text.Command(f"Save Circuit Dir={output_folder}")
+    logger.info("Saved circuit to %s", output_folder)
+
+
+def export_power_flow_results(path: Path):
+    """Export OpenDSS circuit elemement values into a directory."""
+    path.mkdir(exist_ok=True)
+    for export_type, filename in {
+        "currents": "currents.csv",
+        "capacity": "capacity.csv",
+        "loads": "loads.csv",
+        "powers [mva]": "powers.csv",
+        "voltages": "voltages.csv",
+    }.items():
+        dss.Text.Command(f"export {export_type} {path}/{filename}")
+
+
+def compare_power_flow_results(before_path, after_path):
+    """Compare the exported results from two directories.
+
+    Raises
+    ------
+    Exception
+        Raised if the results do not match.
+
+    """
+    before = {x.name: x for x in before_path.iterdir()}
+    after = {x.name: x for x in after_path.iterdir()}
+    assert sorted(before.keys()) == sorted(after.keys())
+    match = True
+    for name in before:
+        if not filecmp.cmp(before[name], after[name]):
+            logger.error("Files before=%s and after=%s do not match", before[name], after[name])
+            match = False
+
+    # TODO: csv comparisons have _mostly_ minor differences.
```

Comment:
This is not a satisfactory comparison. There are minor comparisons that may be fixed with `math.isclose()`. Others would not. We need to resolve this before the PR can be merged.

Reply:
What would be the cause of these differences?

Another TODO: We need to add the Generator object to the list of power conversion elements that can be assigned a profile, in the same category as PVSystem and Storage.